Simplenote/src/androidTest/java/com/automattic/simplenote/utils/TestDriver.kt

```diff
@@ -0,0 +1,95 @@
+package com.automattic.simplenote.utils
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiObjectNotFoundException
+import androidx.test.uiautomator.UiSelector
+import com.automattic.simplenote.R
+import com.automattic.simplenote.uipages.*
+import org.hamcrest.Matchers
+import org.junit.Assert
+import java.lang.Error
+import java.lang.Exception
+
+object TestDriver {
+
+    //Add var for each page object
+    val allNotesHomePage = AllNotesHomePage()
+    val launchPage = LaunchPage()
+    val loginPage = LoginPage()
+    val signUpPage = SignUpPage()
+
+    // Returns True if Login button is visible
+    fun isUserOnLoginPage(): Boolean {
+
+        if (!loginPage.loginPageLoginButton.exists()) {
+            launchPage.loginButton.click()
+            launchPage.loginEmailButton.click()
+        }
+        return loginPage.loginPageLoginButton.exists()
+    }
+
+    /**
+     * Insert the username and password in the LogIn page and does the login
+     */
+    @Throws(UiObjectNotFoundException::class)
+    fun logInWithCredentials(username: String, password: String): Boolean {
+
+        Espresso.onView(
+            Matchers.allOf(
+                ViewMatchers.isDescendantOfA(ViewMatchers.withId(R.id.input_email)),
+                ViewMatchers.withParent(ViewMatchers.withClassName(Matchers.endsWith("android.widget.FrameLayout"))),
+                ViewMatchers.withHint("Email")
+            )
+        )
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+            .perform(ViewActions.replaceText(Constants.ACCOUNT_EMAIL), ViewActions.closeSoftKeyboard())
+
+        Espresso.onView(
+            Matchers.allOf(
+                ViewMatchers.isDescendantOfA(ViewMatchers.withId(R.id.input_password)),
+                ViewMatchers.withParent(ViewMatchers.withClassName(Matchers.endsWith("android.widget.FrameLayout"))),
+                ViewMatchers.withHint("Password")
+            )
+        )
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+            .perform(ViewActions.replaceText(Constants.ACCOUNT_PASSWORD), ViewActions.closeSoftKeyboard())
+
+        loginPage.loginPageLoginButton.click()
+
+        // if True (Open Drawer menu is visible) then login was successful
+        return isOpenDrawerMenuVisible()
+    }
+
+    fun isOpenDrawerMenuVisible(): Boolean {
+
+        if(allNotesHomePage.searchIcon.exists())
+            return true
+        else
+            return false
+    }
+
+    fun logOutUser(): Boolean {
+        return launchPage.signupButton.exists()
+    }
+
+    /**
+     * Tap a button using text
+     */
+    fun tapButton(buttonName: String) {
```

Comment:
I think having the method or parameter named a bit different would remove the necessity of comment for the method. 

Also, since the property by which the button is located is called `text`, I think using this in name would add more clarity to the code.

The third point here it that we're not providing any additional input to `findObject` that will filter only the buttons. In other words, it will work for every object with certain text. Using something like this for the name will reflect what the method does more precisely (just an example):

```suggestion
    fun tapObjectByText(objectText: String) {
```

Reply:
That is perfect, makes sense.